new concept: the-farm - issue 1374 (#1423)
exercism/go: concepts/errors-handling/about.md

# About

> As seen, errors must be the last value returned in a multiple returns function.

Reviewer: ```suggestion
As demonstrated, errors must be the last value returned when a function contains multiple return values.
```

Always return the default value for the type, if returning an error:

```go
func Foo() (int, error) {
  // Ok!
  return 0, errors.New("Error")

  // Don't
  return 10, errors.New("Error")
}
```

> When no errors have to be returned, return `nil`:

Reviewer: ```suggestion
When no errors are encountered, return `nil`:
```

```go
func Foo() (int, error) {
  return 10, nil
}
```

Errors can be checked against `nil`:

```go
file, err := os.Open("./users.csv")
if err != nil {
    // do something with err
    return // or continue
}
// do something with file
```

It is best practise to stop the execution when an error is encountered rather
than using `else`. The example below shows a bad practice:

```go
file, err := os.Open("./users.csv")
if err != nil {

>   // do something wit err

Reviewer: ```suggestion
  // do something with err
```
